capstone-app/src/app/modules/history/wiki-media/wiki-media.component.spec.ts

```diff
@@ -1,49 +1,121 @@
+            langlinks: new Map()
+        }
+
+    const germanLangLinks = new Map<string, Map<string, string>>();
+    const es = new Map<string, string>();
+    es.set("searchQuery", "Stillwater (Oklahoma)");
+    es.set("langName", "Spanisch");
+    es.set("url", "https://es.wikipedia.org/wiki/Stillwater_(Oklahoma)");
+    const en = new Map<string, string>();
+    en.set("searchQuery", "Stillwater, Oklahoma");
+    en.set("langName", "Englisch");
+    en.set("url", "https://en.wikipedia.org/wiki/Stillwater,_Oklahoma");
+    const ca = new Map<string, string>();
+    ca.set("searchQuery", "Stillwater (Oklahoma)");
+    ca.set("langName", "Katalanisch");
+    ca.set("url", "https://ca.wikipedia.org/wiki/Stillwater_(Oklahoma)");
+    germanLangLinks.set('ca', ca);
+    germanLangLinks.set('en', en);
+    germanLangLinks.set('es', es);
+    testWikiGermanServiceResponse.langlinks = germanLangLinks;
+
+    const mockResponse: WikiServiceResult = {
+      title: MockWikiServiceResponse.title,
+      history: MockWikiServiceResponse.history,
+      furtherReading: new Map(Object.entries(MockWikiServiceResponse.furtherReading)),
+      langlinks: new Map(),
+    };
+
+    for (const lang in MockWikiServiceResponse.langlinks){
+      const link = MockWikiServiceResponse.langlinks[lang];
+      const links = new Map<string, string>();
+      links.set('searchQuery', link.searchQuery);
+      links.set('langName', link.langName);
+      links.set('url', link.url);
+      mockResponse.langlinks.set(lang, links);
     }
 
-    const wikiService = jasmine.createSpyObj('WikiResultsService', ['search']);
-    getWikiResponseSpy = wikiService.search.and.returnValue(of(MockWikiResponse));
+    const mockGermanResponse: WikiServiceResult = {
+      title: MockWikiServiceGermanResponse.title,
+      history: MockWikiServiceGermanResponse.history,
+      furtherReading: new Map(Object.entries(MockWikiServiceGermanResponse.furtherReading)),
+      langlinks: new Map(),
+    };
+
+    for (const lang in MockWikiServiceGermanResponse.langlinks){
+      const link = MockWikiServiceGermanResponse.langlinks[lang];
+      const links = new Map<string, string>();
+      links.set('searchQuery', link.searchQuery);
+      links.set('langName', link.langName);
+      links.set('url', link.url);
+      mockGermanResponse.langlinks.set(lang, links);
+    }
+
+    const wikiService = jasmine.createSpyObj('WikiResultsService', ['search', 'searchNewLang']);
+    getWikiResponseSpy = wikiService.search.and.returnValue(of(mockResponse));
```

Comment:
Since getWikiResponseSpy and getWikiLangResponseSpy aren't referenced anywhere, they're not necessary. You can delete where they're defined above and these lines can just be:
```
wikiService.search.and.returnValue(of(mockResponse));
wikiService.searchNewLang.and.returnValue(of(mockGermanResponse));
```